docs/src/userguide/numpy_tutorial.rst

```diff
@@ -138,43 +138,52 @@ Python by using a normal ``import yourmod`` statement.
 The first Cython program
 ==========================
 
-The code below does 2D discrete convolution of an image with a filter (and I'm
-sure you can do better!, let it serve for demonstration purposes). It is both
-valid Python and valid Cython code. I'll refer to it as both
-:file:`convolve_py.py` for the Python version and :file:`convolve_cy.pyx` for the
-Cython version -- Cython uses ".pyx" as its file suffix.
+You can easily execute the code of this tutorial by
+downloading `the Jupyter notebook <https://github.com/cython/cython/blob/master/docs/examples/userguide/numpy_tutorial/numpy_cython.ipynb>`_.
 
-.. literalinclude:: ../../examples/userguide/numpy_tutorial/convolve_py.py
-    :linenos:
+The code below does the equivalent of this function in numpy::
 
-This should be compiled to produce :file:`convolve_cy.so` (for Linux systems). We
+    def compute_np(array_1, array_2, a, b, c):
+        return np.clip(array_1, 2, 10) * a + array_2 * b + c
+
+We'll say that ``array_1`` and ``array_2`` are 2D NumPy arrays of integer type and
+``a``, ``b`` and ``c`` are three Python integers.
+
+This function uses NumPy and is already really fast, so it might be a bit overkill
+to do it again with Cython. This is for demonstration purposes. Nonetheless, we
+will show that we achieve a better speed and memory efficiency than NumPy at the cost of more verbosity.
```

Comment:
This was my main concern when I read that you changed the example. If it's too simple, it might be easier to understand, but it's also less convincing when the speed-up is low and the code overhead very high.
I'll take a deeper look when I have more time.

Reply:
I see your point, we should then do some research to see if it's worth it. 

I'll run some numbers and see the speedups in both cases against a pure python version, and an off-the-shelf library (I'm thinking `scipy.signal.convolve2d` for example).